pkg/controllers/management/feature/feature_handler.go

```diff
@@ -94,14 +104,30 @@ func (h *handler) syncHarvesterFeature(obj *v3.Feature) error {
 	return nil
 }
 
-func (h *handler) toggleHarvesterNodeDriver(harvester string) error {
-	if val := features.GetFeatureByName(harvester).Enabled(); val {
-		m, err := h.nodeDriverController.Cache().Get(harvester)
-		if err != nil {
-			return err
+// syncHarvesterNodeDriver ensures that the Harvester node driver is disabled
+// when the Harvester feature is disabled and that the node driver is enabled,
+// when the Harvester feature is enabled - provided that the node driver
+// exists. If it doesn't exist, the node driver is created.
+func (h *handler) syncHarvesterNodeDriver(feat *v3.Feature) error {
```

Comment:
```suggestion
func (h *handler) syncHarvesterNodeDriver(feature *v3.Feature) error {
```
nit: Personally I'd call this feature. I typically always give function parameters full names unless it's in some sort of library.